src/collisions_updater.cpp

```diff
@@ -0,0 +1,228 @@
+#include <ros/ros.h>
+#include <ros/package.h> // for getting file path for loadng images
+
+#include <moveit/setup_assistant/tools/compute_default_collisions.h>
+#include <moveit/setup_assistant/tools/moveit_config_data.h>
+
+#include <boost/filesystem.hpp>  // for reading folders/files
+#include <boost/program_options.hpp>
+#include <boost/algorithm/string/join.hpp>
+
+#include <fstream>  // for reading in urdf
+#include <streambuf>
+
+namespace fs = boost::filesystem;
+namespace po = boost::program_options;
+
+bool loadFileToString(std::string& buffer, const std::string& path, const std::vector<std::string> &xacro_args){
+
+  if(path.find(".xacro") != std::string::npos){
+        std::string cmd = "rosrun xacro xacro ";
+        for(std::vector<std::string>::const_iterator it = xacro_args.begin(); it != xacro_args.end(); ++it)
+            cmd += *it + " ";
+        cmd += path;
+
+        FILE* pipe = popen(cmd.c_str(), "r");
+        if (!pipe) return false;
+
+        char pipe_buffer[128];
+        while(!feof(pipe)){
+            if(fgets(pipe_buffer, 128, pipe) != NULL)
+                    buffer += pipe_buffer;
+        }
+        pclose(pipe);
+  }else{
+    std::ifstream stream( path.c_str() );
+    if( !stream.good()) return false;
+
+    // Load the file to a string using an efficient memory allocation technique
+    stream.seekg(0, std::ios::end);
+    buffer.reserve(stream.tellg());
+    stream.seekg(0, std::ios::beg);
+    buffer.assign( (std::istreambuf_iterator<char>(stream)), std::istreambuf_iterator<char>() );
+    stream.close();
+  }
+  return true;
+}
+
+class SortableDisabledCollision {
+     const srdf::Model::DisabledCollision dc_;
+     const std::string key_;
+public:
+    SortableDisabledCollision(const srdf::Model::DisabledCollision &dc)
+    : dc_(dc), key_(dc.link1_ < dc.link2_ ? (dc.link1_ + "|" + dc.link2_) : (dc.link2_ + "|" + dc.link1_)) {}
+    operator const srdf::Model::DisabledCollision () const { return dc_; }
+    bool operator < (const SortableDisabledCollision &other) const { return key_ < other.key_; }
+};
+
+class CollisionUpdater{
+    moveit_setup_assistant::MoveItConfigData config_data;
```

Comment:
Good point, the original code used in was parameter, so the underscore was missing. Will fix that.